cli/command/image/build.go

```diff
@@ -250,16 +272,34 @@ func runBuild(dockerCli *command.DockerCli, options buildOptions) error {
 		}
 	}
 
-	ctx := context.Background()
+	if options.stream && dockerfileCtx == nil {
```

Comment:
I'll add a comment. In this case they are equal as nothing has modified `dockerfileCtx` yet but this will change in other comparisons. `dockerfileCtx` doesn't really mean that it has to be from stdin, it means that custom `Dockerfile` source has been set.